### DISCUSSION TITLE
problem with uploading photos and thumbnail generation, permissions

### DISCUSSION THREAD

**Discussion by clix10** [LycheeOrg/Lychee#2809] [Q&A] [answered]
Hello,

So far, I was using "php artisan lychee:sync" command to import photos and that worked fine

But on the new install, I am planing on using drag and drop "upload photo" from UI. And this is not working for me as it should. This is fresh install.

Photos get uploaded, but thumbnails don't get generated. So it looks like this:
![image](https://github.com/user-attachments/assets/91370809-c397-4163-bec3-a0b68777b033)
But files are uploaded OK. I can download back original file from the UI no probme.

In the diagnostic windows, under errors:
warning (below is full diagnostic)
/v**********************lic/uploads/small/89 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/89/d3 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/a1 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/a1/09 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/2b/8a has permissions 2770, but should have 2775
warning
575 more directories with wrong permissions

It is clearly some problem with the permissions. I changed \uploads\ foleder to lychee/lychee 775, but it didn't work. And anytime I upload some pictures, folders that pictures land into change again to www-data/lychee 770.

Also, as error log notes, "/v**********************lic/uploads/..." i don't know if this is "/uploads/..."?

I am runing it on Truenas Electric Eel (supports docker natively), in docker. 

As a template, this docker compose file was used:
https://github.com/LycheeOrg/Lychee-Docker/blob/master/docker-compose.yml

Thank you :)

**Whole diagnostic tab:**

Errors
warning
/v**********************lic/uploads/small/4f has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/4f/27 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/89 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/89/d3 has permissions 2770, but should have 2775
warning
/v**********************lic/uploads/small/74 has permissions 2770, but should have 2775
warning
907 more directories with wrong permissions
warning
APP_URL is still set to default, this will break access to all your images
error
APP_URL (http://lo*****st) does not match the current url (http://192.*********30847).
error
APP_URL (http://lo*****st) does not match the current url (http://192.*********30847).
info
Latest version of PHP is 8.3
info
git (software) is not available.
info
Have you considered supporting Lychee? :)
Info
**Lychee Version (tag):**                      v6.1.2 (9743704) -- Up to date (20 minutes ago).
DB Version:                                6.1.2
composer install:                          --no-dev
APP_ENV:                                   production
APP_DEBUG:                                 false
APP_URL:                                   default
APP_DIR:                                   default
LOG_VIEWER_ENABLED:                        true
VUEJS_ENABLED:                             true
PHOTO_PIPES:                               true
System:                                    Linux
**PHP Version:**                               8.2.26
PHP User agent:                            Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                  Europe/Ljubljana
Max uploaded file size:                    100M
Max post size:                             100M
Chunk size:                                25.60 MB
Max execution time:                        3600
MySQL Version:                             10.11.10-MariaDB-ubu2204
exec() Available:                          yes
Imagick Available:                         1
Imagick Enabled:                           1
Imagick Version:                           1691
GD Version:                                2.3.3
Number of foreign key:                     12 found.

		
Space
Configuration
version:                                   060102
check_for_updates:                         0
sorting_photos_col:                        taken_at
sorting_photos_order:                      ASC
sorting_albums_col:                        max_taken_at
sorting_albums_order:                      ASC
imagick:                                   1
skip_duplicates:                           0
small_max_width:                           0
small_max_height:                          360
medium_max_width:                          1920
medium_max_height:                         1080
lang:                                      en
image_overlay_type:                        desc
default_license:                           none
compression_quality:                       90
grants_full_photo_access:                  1
delete_imported:                           0
mod_frame_enabled:                         1
mod_frame_refresh:                         30
thumb_2x:                                  1
small_2x:                                  1
medium_2x:                                 1
landing_page_enable:                       0
force_migration_in_production:             0
search_public:                             0
hide_version_number:                       0
SL_enable:                                 0
SL_for_admin:                              0
recent_age:                                1
grants_download:                           0
photos_wraparound:                         1
map_display:                               0
zip64:                                     1
map_display_public:                        0
map_provider:                              Wikimedia
force_32bit_ids:                           0
map_include_subalbums:                     0
update_check_every_days:                   3
has_exiftool:                              1
share_button_visible:                      0
import_via_symlink:                        0
has_ffmpeg:                                1
location_decoding:                         0
location_decoding_timeout:                 30
location_show:                             1
location_show_public:                      0
rss_enable:                                0
rss_recent_days:                           7
rss_max_items:                             100
prefer_available_xmp_metadata:             0
editor_enabled:                            1
lossless_optimization:                     0
swipe_tolerance_x:                         150
swipe_tolerance_y:                         250
local_takestamp_video_formats:             .avi|.mov
log_max_num_line:                          1000
unlock_password_photos_with_url_param:     0
nsfw_visible:                              1
nsfw_blur:                                 0
nsfw_warning:                              0
nsfw_warning_admin:                        0
nsfw_banner_override:                      
map_display_direction:                     1
album_subtitle_type:                       oldstyle
upload_processing_limit:                   4
new_photos_notification:                   0
legacy_id_redirection:                     1
zip_deflate_level:                         6
default_album_protection:                  1
album_decoration:                          layers
album_decoration_orientation:              row
allow_username_change:                     1
auto_fix_orientation:                      1
use_job_queues:                            0
random_album_id:                           starred
use_last_modified_date_when_no_exif_date:  0
layout:                                    justified
date_format_photo_thumb:                   M j, Y, g:i:s A e
date_format_photo_overlay:                 M j, Y, g:i:s A e
date_format_sidebar_uploaded:              M j, Y, g:i:s A e
date_format_sidebar_taken_at:              M j, Y, g:i:s A e
date_format_hero_min_max:                  F Y
date_format_hero_created_at:               M j, Y, g:i:s A T
date_format_album_thumb:                   M Y
upload_chunk_size:                         0
nsfw_banner_blur_backdrop:                 0
search_pagination_limit:                   1000
search_minimum_length_required:            4
photo_layout_justified_row_height:         320
photo_layout_masonry_column_width:         300
photo_layout_grid_column_width:            250
photo_layout_square_column_width:          200
photo_layout_gap:                          12
display_thumb_album_overlay:               always
display_thumb_photo_overlay:               hover
default_album_thumb_aspect_ratio:          1/1
use_album_compact_header:                  0
login_button_position:                     left
back_button_enabled:                       0
back_button_text:                          Return to Home
current_job_processing_visible:            1
SA_random_thumbs:                          0
job_processing_queue_visible:              1
login_required:                            0
enable_unsorted:                           1
enable_starred:                            1
enable_recent:                             1
enable_on_this_day:                        1
login_required_root_only:                  1
dark_mode_enabled:                         1
low_quality_image_placeholder:             1
show_keybinding_help_popup:                1
show_keybinding_help_button:               1
disable_se_call_for_actions:               0
enable_se_preview:                         0
slideshow_timeout:                         5
keep_original_untouched:                   1
timeline_photos_enabled:                   1
timeline_photos_public:                    0
timeline_photos_granularity:               day
timeline_photos_order:                     taken_at
timeline_photos_layout:                    square
timeline_photos_pagination_limit:          200
timeline_albums_enabled:                   1
timeline_albums_public:                    0
timeline_albums_granularity:               year
timeline_left_border_enabled:              1
timeline_photo_date_format_year:           Y
timeline_photo_date_format_month:          M Y
timeline_photo_date_format_day:            j M Y
timeline_photo_date_format_hour:           g:i
timeline_album_date_format_year:           Y
timeline_album_date_format_month:          M Y
timeline_album_date_format_day:            j M
hide_nsfw_in_smart_albums:                 1
hide_nsfw_in_frame:                        1
hide_nsfw_in_map:                          1
hide_nsfw_in_search:                       1
search_photos_layout:                      square
hide_nsfw_in_rss:                          1
hide_nsfw_in_timeline:                     1

**Comment by ildyria** [comment, marked as answer]
> It is clearly some problem with the permissions. I changed \uploads\ foleder to lychee/lychee 775, but it didn't work. And anytime I upload some pictures, folders that pictures land into change again to www-data/lychee 770.

No it is a problem of URL. The permission is usually a problem when the folders/images are not created. This is clearly the case here.

Here is the important error: 

> APP_URL ([http://lo*****st](http://lo%2A%2A%2A%2A%2Ast/)) does not match the current url ([http://192.*********30847](http://192.%2A%2A%2A%2A%2A%2A%2A%2A%2A30847/)).

Can you go in your `.env` file probably in the `lychee/conf` folder and see the value of `APP_URL` if it is defined.
If not, I think you can just set it to empty `APP_URL=` that should make Lychee use an empty value for the prefix.

**Comment by clix10** [comment]
Hello,

You are correct. I also checked this, by downloading differnt size images from the menu, and that works fine.

In the default docker-compose file, the environment variable APP_URL is commented out, so the default value is used. As it turns out, the default value is localhost. And this was not working for me. It now works OK, if I set the variable to the actual IP address:port, and it also works OK, if I use APP_URL="" . What is interesting though, If you just write APP_URL=  (without the brackets) docker-compose doesn't recognize it as an empty string and just uses the previously used value.

Thank you, for your quick support.
